scripts/send-bundles-sizes-differences.js

```diff
@@ -0,0 +1,181 @@
+  }
+}
+
+async function getPRs(branch) {
+  const response = await fetch(`https://api.github.com/repos/DataDog/browser-sdk/pulls?head=DataDog:${branch}`, {
+    method: 'GET',
+    headers: {
+      Authorization: `token ${GH_TOKEN}`,
+    },
+  })
+  if (!response.ok) {
+    throw new Error(`HTTP Error Response: ${response.status} ${response.statusText}`)
+  }
+  return response.body ? response.json() : null
+}
+
+function createMessage(difference, resultsBaseQuery, resultsLocalQuery) {
+  let message = '| Bundle | Base Size | Current Size | 𝚫% |\n| --- | --- | --- | --- |\n'
+
+  difference.forEach((diff, index) => {
+    const baseSize = resultsBaseQuery[index].size
+    const localSize = resultsLocalQuery[index].size
+    const sign = diff.percentageChange > 0 ? '+' : ''
+    message += `| ${formatName(diff.name)} | ${baseSize} | ${localSize} | ${sign}${diff.percentageChange}% |\n`
+  })
+
+  return message
+}
+
+async function updateOrAddComment(difference, resultsBaseQuery, resultsLocalQuery, prNumber, commentId) {
+  const message = createMessage(difference, resultsBaseQuery, resultsLocalQuery)
+  const method = commentId ? 'PATCH' : 'POST'
+  const payload = {
+    pr_url: `https://github.com/DataDog/browser-sdk/pull/${prNumber}`,
+    message,
+    header,
+    org: 'DataDog',
+    repo: 'browser-sdk',
+  }
+  const response = await fetch('https://pr-commenter.us1.ddbuild.io/internal/cit/pr-comment', {
+    method,
+    headers: {
+      Authorization: `Bearer ${githubAuthToken}`,
+    },
+    body: JSON.stringify(payload),
+  })
+  if (!response.ok) {
+    throw new Error(`HTTP Error Response: ${response.status} ${response.statusText}`)
+  }
+}
+
+async function getCommentId(prNumber) {
+  const response = await fetch(`https://api.github.com/repos/DataDog/browser-sdk/issues/${prNumber}/comments`, {
+    method: 'GET',
+    headers: {
+      Authorization: `token ${GH_TOKEN}`,
+    },
+  })
+  if (!response.ok) {
+    throw new Error(`HTTP Error Response: ${response.status} ${response.statusText}`)
+  }
+  const comments = await response.json()
+  const targetComment = comments.filter((comment) => comment.body.startsWith('## Bundles Sizes Evolution'))[0]
+  if (targetComment !== undefined) {
+    return targetComment.id
+  }
+}
+
+function loadQuery(commitSha) {
+  return [
+    {
+      name: 'bundles_sizes_rum',
+      queries: `avg:bundle_sizes.rum{commit:${commitSha}}`,
+    },
+    {
+      name: 'bundles_sizes_logs',
+      queries: `avg:bundle_sizes.logs{commit:${commitSha}}`,
+    },
+    {
+      name: 'bundles_sizes_rum_slim',
+      queries: `avg:bundle_sizes.rum_slim{commit:${commitSha}}`,
+    },
+    {
+      name: 'bundles_sizes_worker',
+      queries: `avg:bundle_sizes.worker{commit:${commitSha}}`,
+    },
+  ]
+}
+
+async function queryWithRetry(budget, retries = 4) {
+  const now = Math.floor(Date.now() / 1000)
+  const date = now - 30 * 24 * 60 * 60
```

Comment:
As I used logs2metrics, I think yes we have 30 days retention on those logs

Reply:
oh, it is a metric query and not a logs query 🤦‍♂️
Since we are sending logs in one script and retrieving metrics in this one, it would worth mentioning it, like in the method names for example.

Reply:
What do you think of fetchBundleSizesMetrics instead of queryWithRetry ?

Reply:
It looks great!